src/app/data-model-provider/ActionReturnStatus.cpp

```diff
@@ -119,6 +127,14 @@ ClusterStatusCode ActionReturnStatus::GetStatusCode() const
         return ClusterStatusCode(*err);
     }
 
+    if (const ActionReturnStatus::FixedStatus * status = std::get_if<ActionReturnStatus::FixedStatus>(&mReturnStatus))
+    {
+        if (*status == ActionReturnStatus::FixedStatus::kWriteSuccessNoOp)
+        {
+            return ClusterStatusCode(CHIP_NO_ERROR);
```

Comment:
```suggestion
            return Status::Success;
```

Reply:
I'm not sure if I can change the return value to only `Status::Success`, from what I understand `ClusterStatusCode` can take `Status` in the constructor but it is marked as `explicit`. So probably the compiler won't allow an implicit conversion in here.

Maybe using `ClusterStatusCode(Status::Success)`? It is almost the same as previously but probably makes more clear that this is a "Success"

Reply:
Ah, I see we want a cluster status code... thought we return a direct status. Ok then, code size is probably the same.